test/functional/test_framework/wallet.py

```diff
@@ -71,9 +71,9 @@ def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_sp
         tx.wit.vtxinwit[0].scriptWitness.stack = [CScript([OP_TRUE])]
         tx_hex = tx.serialize().hex()
 
-        txid = from_node.sendrawtransaction(tx_hex)
-        self._utxos.append({'txid': txid, 'vout': 0, 'value': send_value})
-        tx_info = from_node.getmempoolentry(txid)
+        tx_info = from_node.testmempoolaccept([tx_hex])[0]
+        self._utxos.append({'txid': tx_info['txid'], 'vout': 0, 'value': send_value})
+        from_node.sendrawtransaction(tx_hex)
         assert_equal(tx_info['vsize'], vsize)
-        assert_equal(tx_info['fee'], fee)
-        return {'txid': txid, 'wtxid': tx_info['wtxid'], 'hex': tx_hex}
+        assert_equal(tx_info['fees']['base'], fee)
+        return {'txid': tx_info['txid'], 'wtxid': from_node.decoderawtransaction(tx_hex)['hash'], 'hex': tx_hex}
```

Comment:
Unrelated to this PR: What are your thoughts of returning the entire `tx_info` here? There have been a few cases where that would have been nice when I worked with `MiniWallet`. If you think it's useful I could open a PR for that

Reply:
@mjdietzx What fields are you thinking of in particular?

Also I'm pretty sure you can get wtxid with `tx.calc_sha256(with_witness=True)`

Reply:
> Also I'm pretty sure you can get wtxid with tx.calc_sha256(with_witness=True)

sha256 here is is an `int`, not a hex encoding of how txids are represented in Bitcoin (with endian switched for some reason)

Reply:
> If you think it's useful I could open a PR for that

Sure, let's do that in the PR where the return value is needed.